Apps/WebClient/src/ClientApp/src/components/calendar/body.vue

```diff
@@ -175,17 +175,29 @@ export default class CalendarBodyComponent extends Vue {
     @Prop() private isVisible!: boolean;
 
     private eventLimit: number = 4;
-    private firstEventDateOfCurrentMonth: Date | null = null;
 
     private isHovering: boolean = false;
     private hoveringEvent: CalendarEntry | null = null;
     private eventBus = EventBus;
-    private get currentDates() {
-        return this.getCalendar();
+
+    private currentDates: CalendarWeek[] = [];
+
+    @Watch("currentDate")
+    private onCurrentDateUpdate() {
+        this.currentDates = this.generateCalendar();
+
+        if (this.isVisible) {
+            let dateGroup: DateGroup = this.dateGroups.find((d) =>
+                moment(this.currentDate).isSame(d.date, "month")
+            ) as DateGroup;
+            this.eventBus.$emit(
+                EventMessageName.CalendarCurrentDatesUpdated,
```

Comment:
naming? What are the calendar current dates?
Also eventname is CalendarCurrentDate**s**Updated but the passing param is just one date? May be rename the event name to something more meaningful ?

Reply:
I used the enum already in place. I will rename it to be more meaningful